UnityProject/Assets/Scripts/Objects/Disposals/DisposalBin.cs

```diff
@@ -278,19 +286,44 @@ public void PlayerTryClimbingOut(GameObject player)
 
 		#endregion Interactions
 
-		private void StoreItem()
+		// gives the probability of an object falling into the bin. Yes, it's like basketball
+		public void OnFlyingObjectHit(GameObject obj)
+		{
+			var bin = gameObject;
+			if (DMMath.Prob(25))
+			{
+				Chat.AddLocalMsgToChat(obj.ExpensiveName() + " bounces off " + bin.ExpensiveName() + " and doesn't hit it.", bin);
```

Comment:
does this work as intended? Shouldn't it be an actionMessage instead?

Reply:
nvm I think I get it now. So this method adds a message relative to an object instead of a performer, like the action message. Right?

Reply:
We need to set `originatorMessage` for `AddActionMsgToChat()`. Firstly, in my opinion, the `originatorMessage` is not necessary here, and secondly, we don't always know who the originator is

Reply:
yeah, I was confused with the method name. Sorry for making this comment send you a notification.